src/category.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::HashSet;
+use time::Timespec;
+
+use conduit::{Request, Response};
+use conduit_router::RequestParams;
+use pg::GenericConnection;
+use pg::rows::Row;
+
+use {Model, Crate};
+use db::RequestTransaction;
+use util::{RequestUtils, CargoResult, ChainError};
+use util::errors::NotFound;
+
+#[derive(Clone)]
+pub struct Category {
+    pub id: i32,
+    pub category: String,
+    pub created_at: Timespec,
+    pub crates_cnt: i32,
+}
+
+#[derive(RustcEncodable, RustcDecodable)]
+pub struct EncodableCategory {
+    pub id: String,
+    pub category: String,
+    pub created_at: String,
+    pub crates_cnt: i32,
+}
+
+impl Category {
+    pub fn find_by_category(conn: &GenericConnection, name: &str)
+                            -> CargoResult<Option<Category>> {
```

Comment:
If you want to change this to just return `CargoResult<Category>` an internalize the `chain_error(|| NotFound)` that'd also be fine.